spec/datadog/integration_spec.rb

```diff
@@ -21,27 +21,6 @@ def wait_for_tracer_sent
       try_wait_until { Datadog::Tracing.send(:tracer).writer.transport.stats.success > 0 }
     end
 
-    context 'for threads' do
-      let(:original_threads) { Thread.list }
-      let!(:original_threads_inspect) { inspect_threads(original_threads) } # Store result as stack trace will change
-      let(:threads) { Thread.list }
-
-      def inspect_threads(threads)
-        threads.map.with_index { |t, idx| "#{idx}=#{t.backtrace}" }.join(';')
-      end
-
-      it 'closes tracer threads' do
-        start_tracer
-        wait_for_tracer_sent
-
-        shutdown
-
-        expect(threads.size).to eq(original_threads.size),
-          "Expected #{original_threads.size} threads (#{original_threads_inspect}), "\
-                  "got #{threads.size} threads (#{inspect_threads(threads)})"
-      end
-    end
-
```

Comment:
I can't make a suggestion on deleted lines but what if we tested the thread object ids that get created when the tracer starts are no longer present after shutdown, instead of counting how many threads are present? Something like this?

```
    context 'for threads' do
      let(:original_threads) { Thread.list }
      let!(:original_threads_inspect) { inspect_threads(original_threads) }
      let(:tracer_threads) { [] }

      def inspect_threads(threads)
        threads.map.with_index { |t, idx| "#{idx}=#{t.object_id}:#{t.backtrace}" }.join(';')
      end

      subject(:shutdown) { Datadog.shutdown! }

      # Should this replace the :start_tracer let above this context?
      let(:start_tracer) do
        Datadog::Tracing.trace('test.op') {}

        new_threads = Thread.list - original_threads
        tracer_threads.concat(new_threads.map(&:object_id))
      end

      it 'closes tracer threads' do
        start_tracer
        wait_for_tracer_sent

        shutdown

        post_shutdown_threads = Thread.list.map(&:object_id)

        expect(post_shutdown_threads & tracer_threads).to be_empty,
                                                          "Tracer threads not terminated: #{inspect_threads(Thread.list)}"
      end
    end
```

@marcotc - What do you think?

Reply:
That sounds like a smart and reasonable suggestion! Care to take a stab, @ekump? I'd ✅ it!

Reply:
@marcotc sure, I'll create a PR.